### DISCUSSION TITLE
How to get the MapClick data through API?

### DISCUSSION THREAD

**Discussion by wrybread** [weather-gov/api#774] [General]
I'm trying to get the forecast shown on this page through the API:

https://marine.weather.gov/MapClick.php?lat=38.228741371779535&lon=-122.96928405761719

The data under "detailed forecast" at that link.

Using the API I found a few pages about that lat/lon, but not that data:

https://api.weather.gov/points/38.2287,-122.9693

That page lists some URLs but none have the data I'm looking for:

```
"forecast": "https://api.weather.gov/gridpoints/MTR/70,129/forecast",
  "forecastHourly": "https://api.weather.gov/gridpoints/MTR/70,129/forecast/hourly",
  "forecastGridData": "https://api.weather.gov/gridpoints/MTR/70,129",
  "observationStations": "https://api.weather.gov/gridpoints/MTR/70,129/stations",

```
Any tips by chance?

**Comment by dwhitemv25** [comment]
The same (or very similiar) data is under the `forecast` property link. What exactly is missing from there that you need?

**Reply by wrybread** [reply]
Sorry about that, I posted the wrong link, I updated my post. I had clicked on a land coordinate, which was the same data. But a marine point such as this:

https://api.weather.gov/points/38.2287,-122.9693

Gives this forecast:

https://api.weather.gov/gridpoints/MTR/70,129/forecast

Which for example for today reads:

"Sunny. High near 56, with temperatures falling to around 51 in the afternoon. Southwest wind around 5 mph."

But that's missing marine data. The data I'm looking for at this link:

https://marine.weather.gov/MapClick.php?lat=38.228741371779535&lon=-122.96928405761719

That has the marine data for today:

"Variable winds 5 kt or less. Sunny. N swell 0 ft at 0 seconds. Wind waves 1 ft or less."

**Reply by dwhitemv25** [reply]
Marine is a whole different banana. The data you are seeing on marine.weather.gov is extracted from a different set of products than the terrestrial forecasts. 

You want to grab the Coastal Waters Forecast (CWF) product and parse/map/geocode that. For WFO MTR those are available through API at:

https://api.weather.gov/products/types/CWF/locations/MTR

You may find it easier to use the MapClick API, which does all that work for you.
